### PR TITLE
Add `torch.constant.int` and `torch.constant.float`.

### PR DESCRIPTION
**Pull request by silvasean**
- This removes reliance on basicpy.numeric_constant.
- Also, add OpAsmOpInterface to the `torch.constant.none` and
  `torch.constant.str` ops.

### CODE REVIEW

**Comment by silvasean** [comment]
@cathyzhyi PTAL

File: include/npcomp/Dialect/Torch/IR/TorchOps.td [hunk 1]
```diff
@@ -740,4 +741,41 @@ def Torch_ConstantStrOp : Torch_Op<"constant.str",
...
+  let hasFolder = 1;
+}
+
+def Torch_ConstantFloatOp : Torch_Op<"constant.float",
+    [ConstantLike, NoSideEffect, DeclareOpInterfaceMethods<OpAsmOpInterface>]> {
+  let summary = "Materialize a constant `float` value.";
+  let description = [{
+    Note: TorchScript represents `float` as 64-bit floating point values.
+
+    TODO: Add a `!torch.int` type.
```
Comment (cathyzhyi) [inline include/npcomp/Dialect/Torch/IR/TorchOps.td:769]: I think you meant `!torch.float` type

**Comment by cathyzhyi** [comment]
LGTM

File: include/npcomp/Dialect/Torch/IR/TorchOps.td (hunk 1 above)
Comment (silvasean) [reply include/npcomp/Dialect/Torch/IR/TorchOps.td:769]: Thanks!

**Pull request merged**